webapp/lib/controller_tag_helper.dart

```diff
@@ -0,0 +1,59 @@
+part of controller;
+
+class TagManager {
+  static final TagManager _singleton = TagManager._internal();
+
+  List<model.Tag> _tags = <model.Tag>[];
+  List<model.Tag> get tags => _tags;
+
+  TagManager._internal();
+
+  factory TagManager() => _singleton;
+
+  Set<String> get groups => new Set.from(emptyGroups)..addAll(_groupTagsIntoCategories(tags).keys);
```

Comment:
groupNames? groups implies an object to me not a string

Reply:
that's a better name, thanks!